services/web/Dockerfile

```diff
@@ -78,7 +78,8 @@ RUN --mount=type=cache,target=/var/cache/apt,mode=0755,sharing=private \
   set -eux \
   && apt-get update \
   && apt-get install -y --no-install-recommends \
-  build-essential
+  build-essential \
+  libffi-dev
```

Comment:
what is this for?

Reply:
This is what happens when building the Docker image without `libffi-dev` containgin `ffi.h` header:

```
[webserver build 6/6] RUN   --mount=type=bind,source=services/web/server/requirements/_base.txt,target=_base.txt   --mount=type=cache,mode=0755,target=/root/.cache/uv   uv pip install   --requirement _base.txt:
3.360 Resolved 96 packages in 1.43s
4.654 error: Failed to prepare distributions
4.654   Caused by: Failed to fetch wheel: cffi==1.15.0
4.654   Caused by: Build backend failed to build wheel through `build_wheel()` with exit status: 1
4.654 --- stdout:
4.654 running bdist_wheel
4.654 running build
4.654 running build_py
4.654 creating build
4.654 creating build/lib.linux-x86_64-cpython-311
4.654 creating build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/__init__.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/commontypes.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/vengine_cpy.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/setuptools_ext.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/error.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/cparser.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/ffiplatform.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/api.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/pkgconfig.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/backend_ctypes.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/recompiler.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/cffi_opcode.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/lock.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/verifier.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/model.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/vengine_gen.py -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/_cffi_include.h -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/parse_c_type.h -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/_embedding.h -> build/lib.linux-x86_64-cpython-311/cffi
4.654 copying cffi/_cffi_errors.h -> build/lib.linux-x86_64-cpython-311/cffi
4.654 running build_ext
4.654 building '_cffi_backend' extension
4.654 creating build/temp.linux-x86_64-cpython-311
4.654 creating build/temp.linux-x86_64-cpython-311/c
4.654 gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/usr/include/ffi -I/usr/include/libffi -I/root/.cache/uv/builds-v0/.tmpjY4aVT/include -I/usr/local/include/python3.11 -c c/_cffi_backend.c -o build/temp.linux-x86_64-cpython-311/c/_cffi_backend.o
4.654 --- stderr:
4.654 warning: build_py: byte-compiling is disabled, skipping.
4.654 
4.654 c/_cffi_backend.c:15:10: fatal error: ffi.h: No such file or directory
4.654    15 | #include <ffi.h>
4.654       |          ^~~~~~~
4.654 compilation terminated.
4.654 error: command '/usr/bin/gcc' failed with exit code 1
4.654 ---
4.654   Caused by: This error likely indicates that you need to install a library that provides "ffi.h" for cffi==1.15.0
------
Dockerfile:101
--------------------
 100 |     # install only base 3rd party dependencies
 101 | >>> RUN \
 102 | >>>   --mount=type=bind,source=services/web/server/requirements/_base.txt,target=_base.txt \
 103 | >>>   --mount=type=cache,mode=0755,target=/root/.cache/uv \
 104 | >>>   uv pip install \
 105 | >>>   --requirement _base.txt
 106 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c uv pip install   --requirement _base.txt" did not complete successfully: exit code: 2
```